onnxruntime/contrib_ops/cpu/bert/group_query_attention.cc

```diff
@@ -155,8 +159,10 @@ Status GroupQueryAttention<T>::Compute(OpKernelContext* context) const {
       Tensor::InitOrtValue(element_type, TensorShape({batch_size, num_heads_, sequence_length, head_size}), allocator, RotaryQ);
       OrtValue RotaryK;
       Tensor::InitOrtValue(element_type, TensorShape({batch_size, kv_num_heads_, sequence_length, head_size}), allocator, RotaryK);
-      q_input = Q.Get<Tensor>().Data<T>();
-      k_input = K.Get<Tensor>().Data<T>();
+      Q_value_og = Q;
+      K_value_og = K;
```

Comment:
I suggest to remove the following lines
```
Q = RotaryQKV;
...
Q = RotaryQ;
K = RotaryK;

Then change
```
  return ApplyAttention(Q.Get<Tensor>().Data<T>(), packed_qkv ? nullptr : K.Get<Tensor>().Data<T>(),
                        packed_qkv ? nullptr : V.Get<Tensor>().Data<T>(), past_key, past_value, output, present_k, present_v,
                        seqlens_k, parameters, allocator, context);
```
to
```
  return ApplyAttention(q_rotary, packed_qkv ? nullptr : k_rotary,
                        packed_qkv ? nullptr : V.Get<Tensor>().Data<T>(), past_key, past_value, output, present_k, present_v,
                        seqlens_k, parameters, allocator, context);
```